src/library/baseplaylistfeature.cpp

```diff
@@ -473,7 +473,14 @@ void BasePlaylistFeature::slotExportPlaylist() {
     if (file_location.isNull() || file_location.isEmpty()) {
         return;
     }
+    // if the extension was not added add it manually
+    // https://bugreports.qt.io/browse/QTBUG-27186
     QFileInfo fileName(file_location);
+    if (fileName.suffix().isNull() || fileName.suffix().isEmpty()) {
+        QString ext = filefilter.section(".", 1, 1);
+        ext.chop(1);
```

Comment:
Please add a comment what character is chopped here.

Reply:
Extracting this code into a function named `extractFirstSuffixFromFileFilter()` would help.

Reply:
I'll add a debug assertion.

Reply:
> Extracting this code into a function named `extractFirstSuffixFromFileFilter()` would help.

right, for the crate export I assume.
where do you think should `extractSuffixFromFileFilter` go? util/string.h?